sips/sip-032/sip-032-improved-stacking.md

```diff
@@ -0,0 +1,232 @@
+| **Delegation (old)** | Delegating the management of stacking.                                                                                                                   |
+| **Delegation (new)** | Delegating the voting power for proposed blocks.                                                                                                         |
+| **PoX Contract**     | The smart contract that users interact with to lock their tokens.                                                                                        |
+
+## Problem Statement
+
+The current stacking process has two different paths, one for solo Stackers, one for delegated Stackers,
+resulting in a more complex user experience and more complex code.
+
+The process is defined by a prepare phase of 100 bitcoin blocks that is used to find an anchor block
+for the next stacking cycle. Stackers must lock
+their Stacks tokens before the
+prepare phase, i.e. before the 2000th block of the current stacking cycle. Furthermore, Stacks
+tokens are locked for a locking period that always ends at the beginning of a cycle, i.e. after the
+prepare phase. Therefore, the current implementation of Stacking includes a period where Stacks
+tokens are unlocked and ineligible for stacking rewards (Cooldown Cycle).
+
+The current PoX contract allows users to lock Stacks tokens during the prepare phase. This is too late for the next cycle
+and the tokens are locked without earning stacking rewards for one cycle.
+
+The cooldown cycle for unstacking from a pool presents a problem for network decentralization.
+As a Stacker, when I delegate to a signer and that signer does not perform and gets low yield,
+I get penalized for switching. In today's model with cooldown cycles, users get double penalized
+if a signer does not perform. Even if there are penalties for signers not performing / being down,
+the switching costs are too high for users to switch (two weeks' worth of yield).
+
+Furthermore, users cannot change their
+stacking settings (decrease amount, change PoX reward address, etc.) without an unlocked cycle during which users cannot earn
+stacking rewards.
+
+## Proposed Solution
+
+This SIP proposes a new Proof of Transfer (PoX) contract without
+the flow for solo stacking and moves responsibilities from pool operators to signers who were introduced in the Nakamoto upgrade. It also
+defines a new locking mechanism that allows users to switch from one signer to another with a single contract call and without a cooldown period.
+
+# Specification
+
+Applying these upgrades to the Stacks blockchain requires a consensus-breaking network upgrade,
+in this case, a hard fork. Like other such changes, this will require a new Stacks epoch.
+In this SIP, we will refer to this new epoch as Stacks 3.2.
+
+## Delegated Stacking only
+
+The following PoX contract functions shall be removed:
+
+- stack-stx
+- stack-increase
+- stack-extend
```

Comment:
Instead of removing them, could they instead wrap calls to the delegation functions?  For example, `stack-stx` could be re-implemented to delegate STX to the caller, and have the caller stack and aggregate-commit them.